### DISCUSSION TITLE
What is the process for generating 10.3 bindings?

### DISCUSSION THREAD

**Discussion by flatheadmill** [grafana/grafonnet#167] [General]
Seems like it's at 10.2 and when I apply a dashboard, then save the dashboard in 10.3 there is a decent sized diff. I know it's only a couple weeks old. Should bleeding edge users just run `generate.sh` themselves?

**Comment by Duologic** [comment]
> then save the dashboard in 10.3 there is a decent sized diff

The saved JSON output is likely more verbose regardless of the Grafana version, it includes lots of default values that would get set anyway, Grafonnet only generates the JSON that is required without these defaults.

> What is the process for generating 10.3 bindings?

Today the process is a bit tedious and requires some fiddling on step 1:

1. generate the JSON Schemas in [Grok](https://github.com/grafana/grok/) (Update go.mod, update Makefile and create a PR)
2. add entry for 10.3 to the Makefile in Grafonnet
3. generate it, create PR, merge
4. update Makefile 'latest' 
5. generate it, create PR, merge ( mostly docs changes probably)

**Comment by Duologic** [comment]
Bindings produced in https://github.com/grafana/grafonnet/pull/175
